src/core/trace_core.mli

```diff
@@ -34,8 +34,9 @@ val with_span :
     see {!enter_manual_span}.
 *)
 
-val add_data_to_span : span -> (string * user_data) list -> unit
-(** Add structured data to the given span.
+val add_data : (string * user_data) list -> unit
+(** Add structured data to the current span (the ambient {!with_span}).
+    Behavior is not specified if there is no current span.
```

Comment:
We definitely shouldn't leave behaviour unspecified, in this interface — the entire _purpose_ of this interface is to fully specify, so that all collectors have to conform to eachother.

This is basically a discussion of 'before a span has been opened', yes? Perhaps the simplest two options are:

1. Dictate that it's always silently dropped;
2. dictate that the collector store any data pre-span, then add it to the first span seen;
3. dictate that the collector call `internal_message` to report a warning (as mentioned in the other thread.)

Reply:
I think collectors will have various constraints, and this is plain and simply a bug. I think it should be encouraged (via docs?) that `add_data` is used in a scope very close to the one where the span is created, so it should just not happen that it's used outside of any scope.

E.g in TEF it's just going to be dropped because there is nowhere where the data could go; but in OTEL it could be turned into a log, for example, possibly. In Tracy it's dropped _anyway_.